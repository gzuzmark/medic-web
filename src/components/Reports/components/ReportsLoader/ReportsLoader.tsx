import * as React from "react";
import { Heading2 } from '../../../../common/ConsoleText';
import Loader from "../../../../common/Loader/Loader";
import './ReportsLoader.scss'

interface IPropsReportsLoader {
    loading: boolean;
    center: boolean;
}

const ReportsLoader: React.StatelessComponent<IPropsReportsLoader> = (props) => {
    if (!props.loading) {
        return null;
    }
    return (
        <div className={`ReportsLoader ${props.center && 'ReportsLoader--center'}`}>
            <Loader top={50} height={100} style={{top: -37}}/>
            <Heading2 style={{fontWeight: 'bold', marginLeft: 30}}>
                {props.children}
            </Heading2>
        </div>
    )
};


export default ReportsLoader;