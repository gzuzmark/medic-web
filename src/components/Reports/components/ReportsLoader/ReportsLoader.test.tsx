import { shallow } from 'enzyme';
import 'jest-localstorage-mock';
import * as React from 'react';
import ReportsLoader from './ReportsLoader';

describe('ReportsLoader Test',() => {
    let props: any;
    let mountedReportsLoader: any;
    const getComponent = () => {
        if (!mountedReportsLoader) {
            mountedReportsLoader = shallow(
                <ReportsLoader {...props}>Caja de Mensajes</ReportsLoader>
            );
        }
        return mountedReportsLoader;
    };

    beforeEach(() => {
        props = {
            center: false,
            loading: true
        };
        mountedReportsLoader = undefined;
    });

    it("render: render ReportsLoader", () => {
        const component = getComponent();
        expect(component).toMatchSnapshot();
    });

    it("render: render ReportsLoader should have class center", () => {
        props = {
            center: true,
            loading: true
        };
        const component = getComponent().find('.Reports-loader--center');
        expect(component.length).toBe(1);
    });

    it("render: render ReportsLoader should be null", () => {
        props = {
            center: false,
            loading: false
        };
        const component = getComponent().find('.ReportsLoader');
        expect(component.length).toBe(0);
    });

});