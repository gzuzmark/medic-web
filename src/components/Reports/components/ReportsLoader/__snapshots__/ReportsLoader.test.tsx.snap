// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportsLoader Test render: render ReportsLoader 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReportsLoader
    center={false}
    loading={true}
  >
    Caja de Mensajes
  </ReportsLoader>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Loader
          height={100}
          style={
            Object {
              "top": -37,
            }
          }
          top={50}
        />,
        <styled.h2
          style={
            Object {
              "fontWeight": "bold",
              "marginLeft": 30,
            }
          }
        >
          Caja de Mensajes
        </styled.h2>,
      ],
      "className": "ReportsLoader false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "height": 100,
          "style": Object {
            "top": -37,
          },
          "top": 50,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Caja de Mensajes",
          "style": Object {
            "fontWeight": "bold",
            "marginLeft": 30,
          },
        },
        "ref": null,
        "rendered": "Caja de Mensajes",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Loader
            height={100}
            style={
              Object {
                "top": -37,
              }
            }
            top={50}
          />,
          <styled.h2
            style={
              Object {
                "fontWeight": "bold",
                "marginLeft": 30,
              }
            }
          >
            Caja de Mensajes
          </styled.h2>,
        ],
        "className": "ReportsLoader false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "height": 100,
            "style": Object {
              "top": -37,
            },
            "top": 50,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Caja de Mensajes",
            "style": Object {
              "fontWeight": "bold",
              "marginLeft": 30,
            },
          },
          "ref": null,
          "rendered": "Caja de Mensajes",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
